pilot/pkg/kube/inject/inject.go

```diff
@@ -481,20 +482,31 @@ func directory(filepath string) string {
 	return dir
 }
 
-func injectionData(sidecarTemplate, version string, deploymentMetadata *metav1.ObjectMeta, spec *corev1.PodSpec,
+func flippedContains(needle, haystack string) bool {
+	return strings.Contains(haystack, needle)
+}
+
+func injectionData(sidecarTemplate, valuesConfig, version string, deploymentMetadata *metav1.ObjectMeta, spec *corev1.PodSpec,
 	metadata *metav1.ObjectMeta, proxyConfig *meshconfig.ProxyConfig, meshConfig *meshconfig.MeshConfig) (
 	*SidecarInjectionSpec, string, error) {
 	if err := validateAnnotations(metadata.GetAnnotations()); err != nil {
 		log.Errorf("Injection failed due to invalid annotations: %v", err)
 		return nil, "", err
 	}
 
+	values := map[string]interface{}{}
+	if err := yaml.Unmarshal([]byte(valuesConfig), &values); err != nil {
```

Comment:
encoded as JSON in the installer, but unmarshaled as yaml in the injector?

Reply:
Good point. I encoded as json in the installer because its easier to not have to deal with yaml quirks and spacing issues etc. Yaml is a superset so this works, but maybe this should change. Although if someone does provide a custom values file, they will pass yaml - very very niche case though